src/app/collective-rewards/shared/actions.ts

```diff
@@ -0,0 +1,67 @@
+'use server'
+
+import { makeClient } from '@/shared/components/ApolloClient'
+import { gql as apolloGQL } from '@apollo/client'
+import { Address } from 'viem'
+
+type CycleData = {
+  id: string
+  rewardsERC20: string
+  rewardsRBTC: string
+}
+
+type BackerRewardPercentageData = {
+  id: string
+  next: string
+  previous: string
+  cooldownEndTime: string
+}
+
+type BuilderData = {
+  id: Address
+  backerRewardPercentage: BackerRewardPercentageData
+  rewardShares: string
+  totalAllocation: string
+}
+
+type Response = {
+  builders: BuilderData[]
+  cycles: CycleData[]
+}
+
+const apolloQuery = apolloGQL`
+  query AbiMetricsData {
+    builders(
+      where: { state_: { kycApproved: true, communityApproved: true, initialized: true, selfPaused: false } }
+      orderBy: totalAllocation
+      orderDirection: desc
+    ) {
+      id
+      totalAllocation
+      backerRewardPercentage {
+        id
+        next
+        previous
+        cooldownEndTime
+      }
+    }
+    cycles(first: 1, orderBy: id, orderDirection: desc) {
+      id
+      rewardsERC20
+      rewardsRBTC
+    }
+  }
+`
+
+const fetchCrTheGraphEndpoint = `${process.env.THE_GRAPH_URL}/${process.env.THE_GRAPH_API_KEY}/${process.env.THE_GRAPH_ID}`
```

Comment:
This is built in two places, maybe we can move it two a single place,  but maybe can be updated when we update the RBI.